pallets/identity/src/auth.rs

```diff
@@ -187,7 +196,24 @@ impl<T: Config> Module<T> {
         }
 
         // Run custom per-type validation and updates.
-        accepter(auth.authorization_data, auth.authorized_by)?;
+        let res = accepter(auth.authorization_data, auth.authorized_by);
+
+        if res.is_err() {
+            let mut updated_auth = Self::ensure_authorization(target, auth_id)?;
+            // decrement
+            updated_auth.count -= 1;
```

Comment:
In Rust sub/add/etc... operators that underflow/overflow will panic.  Please make sure to use `saturating_sub` instead.

```suggestion
            // decrement
            updated_auth.count = updated_auth.count.saturating_sub(1);
```